ComputationalArt.py

```diff
@@ -0,0 +1,256 @@
+    """ Evaluate the random function f with inputs x,y
+        Representation of the function f is defined in the assignment writeup
+
+        f: the function to evaluate
+        x: the value of x to be used to evaluate the function
+        y: the value of y to be used to evaluate the function
+        returns: the function value
+
+    """
+    if f[0] == "x":
+        return x
+    if f[0] == "y":
+        return y
+    return f[0](*[evaluate_random_function_real(g,x,y) for g in f[1:]])
+
+
+def build_random_function_imaginary(min_depth,max_depth,f_list):
+    """ This function builds an imaginary function that can be used for HSV
+
+    """
+    g = f_list
+    if max_depth == 1:
+        return ["I"]
+    elif min_depth == 1:
+        functionDict = allFunctions_imaginary
+        choices = f_list+["I"]
+    else:
+        functionDict = functions_imaginary
+        choices = f_list
+    functionName = random.choice(choices)
+    function = functionDict[functionName]
+    if function != "I":
+        nParams = len(signature(function).parameters)
+        params = [build_random_function_imaginary(min_depth-1,max_depth-1,f_list) for _ in range(nParams)]
+        return [function] + params
+    else:
+        return [function]
+
+
+def evaluate_random_function_imaginary(f, c):
+    """This function evaluates the randomly generated imaginary function.
+    It first goes through in a a+bi manner evaluating based on real and imaginary
+    components. It outputs the real component and the imaginary one.
+
+    NOT NOWx is a complex number a+bi where a =x and b = y
+    y is a complex number a+bi where a = y and b =x
+    """
+    if f[0] == "I":
+        return c
+    return f[0](*[evaluate_random_function_imaginary(g,c) for g in f[1:]])
+
+
+def remap_interval(val,
+                   input_interval_start,
+                   input_interval_end,
+                   output_interval_start,
+                   output_interval_end):
+    """ Given an input value in the interval [input_interval_start,
+        input_interval_end], return an output value scaled to fall within
+        the output interval [output_interval_start, output_interval_end].
+
+        val: the value to remap
+        input_interval_start: the start of the interval that contains all
+                              possible values for val
+        input_interval_end: the end of the interval that contains all possible
+                            values for val
+        output_interval_start: the start of the interval that contains all
+                               possible output values
+        output_inteval_end: the end of the interval that contains all possible
+                            output values
+        returns: the value remapped from the input to the output interval
+
+        >>> remap_interval(0.5, 0, 1, 0, 10)
+        5.0
+        >>> remap_interval(5, 4, 6, 0, 2)
+        1.0
+        >>> remap_interval(5, 4, 6, 1, 2)
+        1.5
+    """
```

Comment:
Very good docstring. It has everything in needs: description of this function, what input it expects, and what output it produces.